File: backend/employee-service/src/main/java/com/nexushr/service/PerformanceService.java
package com.nexushr.service;

import com.nexushr.dto.performance.PerformanceConfigurationDto;
import com.nexushr.dto.performance.PerformanceGenerateRequest;
import com.nexushr.dto.performance.PerformanceRecordDto;
import com.nexushr.dto.performance.PerformanceReportDto;

import java.util.List;

public interface PerformanceService {
    PerformanceConfigurationDto getConfiguration();
    PerformanceConfigurationDto updateConfiguration(PerformanceConfigurationDto request);

    void generateMonthlyPerformance(PerformanceGenerateRequest request, String generatedBy, String authHeader);
    void publishPerformance(PerformanceGenerateRequest request);

    PerformanceRecordDto getMyPerformance(Long employeeId, Integer year, Integer month, boolean requirePublished);
    List<PerformanceRecordDto> getMyPerformanceHistory(Long employeeId, boolean requirePublished);

    List<PerformanceRecordDto> getTeamPerformance(Long managerId, Integer year, Integer month, boolean requirePublished);
    List<PerformanceRecordDto> getAllPerformance(Integer year, Integer month);
    
    PerformanceReportDto getPerformanceReport(Integer year, Integer month);
}

File: frontend/src/services/performance.service.ts
import api from './api';
import type { 
  PerformanceConfiguration, 
  PerformanceRecord, 
  PerformanceGenerateRequest, 
  PerformanceReport 
} from '../types/performance.types';

export const performanceService = {
  // Config
  getConfiguration: () => api.get<PerformanceConfiguration>('/employees/performance/configuration'),
  updateConfiguration: (data: PerformanceConfiguration) => api.put<PerformanceConfiguration>('/employees/performance/configuration', data),

  // Actions
  generatePerformance: (request: PerformanceGenerateRequest) => api.post<string>('/employees/performance/generate', request),
  publishPerformance: (request: PerformanceGenerateRequest) => api.post<string>('/employees/performance/publish', request),
  lockPerformance: (request: PerformanceGenerateRequest) => api.post<string>('/employees/performance/lock', request),

  // Queries
  getMyPerformance: (year: number, month: number) => api.get<PerformanceRecord>(`/employees/performance/me?year=${year}&month=${month}`),
  getMyPerformanceHistory: () => api.get<PerformanceRecord[]>('/employees/performance/history/me'),
  getTeamPerformance: (year: number, month: number) => api.get<PerformanceRecord[]>(`/employees/performance/team?year=${year}&month=${month}`),
  getAllPerformance: (year: number, month: number) => api.get<PerformanceRecord[]>(`/employees/performance/all?year=${year}&month=${month}`),
  getEmployeePerformanceHistory: (employeeId: number) => api.get<PerformanceRecord[]>(`/employees/performance/history/${employeeId}`),
  getPerformanceReport: (year: number, month: number) => api.get<PerformanceReport>(`/employees/performance/report?year=${year}&month=${month}`),
  getGeneratedMonths: () => api.get<string[]>('/employees/performance/generated-months')
};
